src/ui/public/vis_maps/visualizations/tile_map.js

```diff
@@ -98,12 +98,13 @@ export default function TileMapFactory(Private) {
 
       const params = _.assign({}, _.get(this._chartData, 'geoAgg.vis.params'), uiStateParams);
 
+      const tooltipFormatter = this.handler.visConfig.get('addTooltip') ? this.tooltipFormatter : null;
```

Comment:
you could leave tooltipFormatter as is, as addTooltip parameter gets passed down to the map and you could just check that.